### PR TITLE
Add Contributing Guidelines

### PR DESCRIPTION
**Pull request by phoenixpereira**
### Description
Added a basic contributing guidelines and updated the README file.

### Changes Made
- Customised the README file to suit our repo.
- Added contributing guidelines with instructions for reporting issues and contributing changes.

### Related Issues
N/A

### Additional Notes
Contributing guidelines are useful for new people who want to contribute to the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-pnpm run db:push
+pnpm db:push
 ```
 
-4. Run dev server
+5. Then run the development server
 
 ```bash
-pnpm run dev
+pnpm dev
```
Comment (rayokamoto) [inline README.md:32]: It should be `pnpm run dev`

File: README.md [hunk 2]
```diff
@@ -1,23 +1,39 @@
 # CS Club Website
 
-## Development
+---
 
-1. Install dependencies
+This is the official repository for the University of Adelaide Computer Science Club (CS Club) website. It is built using [Next.js](https://nextjs.org/), [React](https://reactjs.org/), [TypeScript](https://www.typescriptlang.org/), and [Tailwind CSS](https://tailwindcss.com/).
```
Comment (rayokamoto) [inline README.md:5]: I don't know if it's necessary to list out the tech stack we're using

File: README.md [hunk 3]
```diff
@@ -1,23 +1,39 @@
 # CS Club Website
 
-## Development
+---
```
Comment (rayokamoto) [inline README.md:3]: This separating line isn't needed in my opinion

File: README.md [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
 ```
 
-2. Rename `.env.local.example` to `.env.local` and edit it
+2. Install the dependencies.
+```bash
+pnpm i
+```
 
-3. Initialize database
+3. Set up the keys by copying `.env.local.example` to a new file `.env.local` and replace the placeholder keys with the actual keys. Please contact the open source officers on the CS Club Discord to get these keys.
```
Comment (rayokamoto) [inline README.md:22]: We wouldn't be giving out the keys to every single contributor though?

File: README.md [hunk 5]
```diff
@@ -1,23 +1,39 @@
...
 
+4. Initialise the database
 ```bash
-pnpm run db:push
+pnpm db:push
```
Comment (rayokamoto) [inline README.md:26]: Should be `pnpm run db:push`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
+# Contributing Guidelines
+
+You can contribute to the University of Adelaide Computer Science Club (CS Club) website with issues, PRs and testing of PRs. We appreciate your efforts to make our website better. Please read the entire documentation before continuing to make sure you understand all the processes.
```
Comment (rayokamoto) [inline CONTRIBUTING.md:3]: >  website with issues, PRs and testing of PRs.

Add comma between "PRs and"

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+### Suggested Workflow
+
+We recommend you to use the following workflow:
+
+1. Find an existing issue or create a new one.
+    - Seek feedback from the maintainers and the community that your proposed change is a good one.
+    - Clearly state whether you plan on implementing it yourself or not. If this is the case, you can request that the issue be assigned to you. Note: The person who filed the issue and the person who implements it don't have to be the same.
+2. Create a personal fork of the repository using GitHub (if you don't already have one).
+3. In your fork, create a branch based of main (`git checkout -b mybranch`).
```
Comment (rayokamoto) [inline CONTRIBUTING.md:48]: Maybe specify that `mybranch` be replaced with a relevant branch name?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+4. Create and commit your changes to your branch.
+    - Commit messages should follow the guidelines in [Commit Messages](#commit-messages)
+6. Run the linting and prettier commands with `pnpm lint` and `pnpm format` to ensure your code follows our code quality standards.
```
Comment (rayokamoto) [inline CONTRIBUTING.md:51]: The commands should be `pnpm run lint` and `pnpm run format`. Although as far as I know, there is a git hook that does both automatically when pushing to GitHub? Am I correct @jsun969?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+7. Create a pull request (PR) against the compsci-adl/website repository's **main** branch.
```
Comment (rayokamoto) [inline CONTRIBUTING.md:52]: - `compsci-adl/website` can be linked
- change from **main** to `main`

I think reword this to:
> Create a pull request (PR) to merge your changes into the [compsci-adl/website](https://github.com/compsci-adl/website) repository's `main` branch.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+    - State in the description what issue or improvement your change is addressing.
+    - Check if all the Continuous Integration checks are passing. Refer to [Actions](https://github.com/compsci-adl/website/actions) to check for errors.
+8. Wait for feedback or approval of your changes from the maintainers
+9. When the maintainers have signed off, and all checks are green, your PR will be merged.
```
Comment (rayokamoto) [inline CONTRIBUTING.md:56]: Mention there being no merge conflicts as well

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+- `ci`: Changing the configuration of continuous integration (i.e.
+  Github actions)
+- `chore`: Changes that do not fit into any of the above
+  categories
+
+If you are interested in the detailed specification you can visit
+<https://www.conventionalcommits.org/>
+
+### Code Style
+We follow the [AirBnb Javascript Style Guide](https://github.com/airbnb/javascript) and the [Airbnb React/JSX Style Guide](https://github.com/airbnb/javascript/tree/master/react). Please take the time to familiarise yourself with these guides.
```
Comment (rayokamoto) [inline CONTRIBUTING.md:83]: change "AirBnb" to "Airbnb"

**Review by rayokamoto — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (phoenixpereira) [reply README.md:32]: `pnpm dev` works the same as `pnpm run dev`.

File: README.md (hunk 2 above)
Comment (phoenixpereira) [reply README.md:5]: It's more so for new contributors who are interested in contributing to the project. Although, it might be better to put this info in `CONTRIBUTING.md`.

File: README.md (hunk 1 above)
Comment (rayokamoto) [reply README.md:32]: For me it doesn't

File: CONTRIBUTING.md (hunk 11 above)
Comment (jsun969) [reply CONTRIBUTING.md:83]: Actually we don't. I checked this style guide and ig it's too old for nextjs 14 and react 18.

File: README.md (hunk 5 above)
Comment (phoenixpereira) [reply README.md:26]: Same as before, `pnpm db:push` works the same as `pnpm run db:push`.

File: README.md (hunk 1 above)
Comment (jsun969) [reply README.md:32]: I agree to add a `run` for each command.

File: CONTRIBUTING.md (hunk 8 above)
Comment (phoenixpereira) [reply CONTRIBUTING.md:51]: I think he implemented the git hook for it, but at least for me, it doesn't seem to work properly since I have to run them manually for it to check the code.

File: README.md (hunk 1 above)
Comment (phoenixpereira) [reply README.md:32]: Huh, okay I'll change it then.

File: CONTRIBUTING.md (hunk 11 above)
Comment (rayokamoto) [reply CONTRIBUTING.md:83]: Yeah, I think I said we should somewhat follow the style guide but not completely.

File: CONTRIBUTING.md (hunk 8 above)
Comment (jsun969) [reply CONTRIBUTING.md:51]: We have format githook

File: README.md [hunk 12]
```diff
@@ -1,23 +1,39 @@
...
-## Development
+---
 
-1. Install dependencies
+This is the official repository for the University of Adelaide Computer Science Club (CS Club) website. It is built using [Next.js](https://nextjs.org/), [React](https://reactjs.org/), [TypeScript](https://www.typescriptlang.org/), and [Tailwind CSS](https://tailwindcss.com/).
+
+## Getting Started
+
+To get started, please follow these steps: 
+1. Clone the repository and open the folder.
```
Comment (jsun969) [inline README.md:10]: Remove this step. I assume it's dumplicated with contributing guide. Also, it's recommanded to use ssh to clone a git repo.

File: CONTRIBUTING.md (hunk 11 above)
Comment (phoenixpereira) [reply CONTRIBUTING.md:83]: What would be the best way to word this?

File: README.md (hunk 4 above)
Comment (phoenixpereira) [reply README.md:22]: Depends on the features they're working on I guess since the account related stuff would require keys. Perhaps we could tell them to make their own Clerk account with their own keys and change needed settings on it?

File: CONTRIBUTING.md
Comment (jsun969) [inline CONTRIBUTING.md:1]: This is a great and complete contributing guide, but I suggest we should show the contributor more commands than text. cuz I assume contributors don't like to read a lot of words before coding.

Here is my contributing guide:
https://github.com/jsun969/just-submit/blob/main/CONTRIBUTING.md

**Review by jsun969 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 11 above)
Comment (jsun969) [reply CONTRIBUTING.md:83]: Maybe just remove this for now?

**Comment by phoenixpereira** [comment]
Do we want to add a license to the project?

**Comment by rayokamoto** [comment]
We should probably add a license, yes. What licenses are you thinking of?

Some common licenses:
https://choosealicense.com/

**Comment by jsun969** [comment]
AGPL v3 or MIT ?

**Comment by phoenixpereira** [comment]
Probably MIT

**Comment by rayokamoto** [comment]
Happy to go with MIT

File: README.md (hunk 4 above)
Comment (rayokamoto) [reply README.md]: That might be the way to go.

Comment (phoenixpereira) [reply README.md]: Should we just say to make their own keys then?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+Project maintainers will merge changes that both improve the project and meet our code quality standards. 
+
+### Suggested Workflow
+
+We recommend you to use the following workflow:
+
+1. Fork and clone the repo to your computer.
+```bash
+git clone https://github.com/YOUR_GITHUB_USERNAME/website.git
```
Comment (jsun969) [inline CONTRIBUTING.md:46]: sry this is a mistake in my projext
```suggestion
git clone git@github.com:compsci-adl/website.git
```

File: CONTRIBUTING.md (hunk 8 above)
Comment (phoenixpereira) [reply CONTRIBUTING.md]: When you lint, some warnings don't get automatically fixed, so it's beneficial to manually lint as well.

File: LICENSE [hunk 14]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 The University of Adelaide Computer Science Club
```
Comment (jsun969) [inline LICENSE:3]:
```suggestion
Copyright (c) 2023-present The University of Adelaide Computer Science Club
```

File: CONTRIBUTING.md (hunk 13 above)
Comment (rayokamoto) [reply CONTRIBUTING.md:46]: I agree that using ssh is better than http, especially for private repos and when committing, I think http is just easier for beginners cloning a public repository. Because with ssh, you'd need to set up ssh keys and whatnot.

File: README.md [hunk 15]
```diff
@@ -1,23 +1,34 @@
...
+
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+## Contributing
+
+We welcome contributions to enhance the CS Club Website! If you find any issues, have suggestions, or want to request a feature, please follow our [Contributing Guidelines](CONTRIBUTING.md).
+
+## License
+This project is licensed under the MIT License. 
+See the [LICENSE](LICENSE) file for details.
```
Comment (rayokamoto) [inline README.md:34]:
```suggestion
See [LICENSE](LICENSE) for details.
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+git push -u origin your-relevant-branch-name
+```
+7. Once you are happy with your changes, create a pull request (PR) with a relevant title and description to merge your changes into the [compsci-adl/website](https://github.com/compsci-adl/website) repository's `main` branch.
+8. Wait for feedback and approval of your changes from the maintainers
+9. When the maintainers have signed off, all checks are green, and there are no merge conflicts, your PR will be merged.
+
+### Commit Messages
+Before you create a PR, please check whether your commits comply with the commit conventions used for this repository.
+When you create a commit, please follow the convention
+`<type>[optional scope]: <description>` in your commit message while using one of
```
Comment (rayokamoto) [inline CONTRIBUTING.md:76]: Maybe provide an example of what this might look like, e.g.:

```
feat(db): Add database schema

ci: Add prettier to CI workflow
```

**Review by rayokamoto — CHANGES_REQUESTED** [review]

**Review by jsun969 — APPROVED** [review]
LGTM

**Review by rayokamoto — APPROVED** [review]
